server/controller/profile.js

```diff
@@ -91,6 +94,90 @@ module.exports = class ProfileController extends BaseController {
 		}
 	}
 
+	async handleUpdateAccount(req, res, _next) {
+		try {
+			const { fName, lName } = req.body;
+
+			if (!res.locals.user) {
+				return this.handleError(res, 403, 'No user id in request');
+			}
+
+			const user = await this.#userRepository.getUserByObjectId(res.locals.user._id);
+
+			if (!user) {
+				return this.handleError(res, 404, 'User could not be found');
+			}
+
+			await this.#userRepository.updateUserById(user._id, { fName, lName });
+
+			res.status(200).send({
+				success: true,
+				error: null,
+				data: { fName, lName },
+			});
+		} catch (err) {
+			return this.handleError(res, 400, err);
+		}
+	}
+
+	async handleUpdateAgency(req, res, _next) {
+		try {
+			const {
+				agencyBio,
+				agencyPhone,
+				agencyWebsite,
+				address1,
+				address2,
+				city,
+				state,
+				country,
+				zipcode,
+			} = req.body;
+
+			if (!res.locals.user) {
+				return this.handleError(res, 403, 'No user id in request');
+			}
+
+			const accountManager = await this.#agencyRepository.getAgencyByUserId(
```

Comment:
You want to call `this.#userRepository.getUserByObjectId` right? Since you want the account manager's object to get the `_id` from him

Reply:
I want to get the agency the user (account manager) belongs to, so that I can update their agency information.

`
const AgencySchema = new Schema(
	{
		agencyName: {
			type: String,
		},
		agencyWebsite: {
			type: String,
		},
		agencyPhone: {
			type: String,
		},
		**accountManager: {
			type: Schema.Types.ObjectId,
			ref: 'User',
		},**
		agencyBio: {
			type: String,
		},
		agencyAddress: {
			address1: { type: String },
			address2: { type: String },
			city: { type: String },
			state: { type: String },
			country: { type: String },
			zipcode: { type: String },
		},
		childrenUnderCare: {
			type: Number,
		},
		childrenAgeRange: {
			type: String,
		},
		agencyProfileImage: {
			type: String,
		},
		joined: {
			type: Date,
			default: Date.now,
		},
		isVerified: {
			type: Boolean,
			default: false,
		},
	},
	{
		collection: 'agencies',
	},
);
`

Reply:
yes I noticed that, but, you actually don't need to get the user either since you already have the user._id in the response object `res.locals.user._id`

and the `updateAgency` function takes as first parameter the user id

Reply:
Ok fixed the param issue